### PR TITLE
Add condition to opa components call

### PR DESCRIPTION
**Pull request by kgroschoff**
### What this PR does / why we need it
Added condition to opa components call.

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by kgroschoff** [comment]
Maybe it is just a coincidence, but today I had problems to create machine deployments, especially when opa is enabled. It seems like the endpoints for constraints & gatekeeper config are called too early, which then breaks the machine deployment creation. 

PTAL @maciaszczykm / @floreks

File: src/app/cluster/cluster-details/component.ts [hunk 1]
```diff
@@ -403,6 +403,10 @@ export class ClusterDetailsComponent implements OnInit, OnDestroy {
     );
   }
 
+  isHavingMD(): boolean {
```
Comment (maciaszczykm) [inline src/app/cluster/cluster-details/component.ts:406]: s/isHavingMD/hasMachineDeployments

File: src/app/cluster/cluster-details/constraints/component.ts [hunk 2]
```diff
@@ -97,11 +98,11 @@ export class ConstraintsComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   isLoadingData(data: Constraint[]): boolean {
-    return _.isEmpty(data) && !this.isClusterRunning;
+    return _.isEmpty(data) && !this.isClusterRunning && !this.hasMachineDeployments;
   }
 
   hasNoData(data: Constraint[]): boolean {
-    return _.isEmpty(data) && this.isClusterRunning;
+    return _.isEmpty(data) && this.isClusterRunning && this.hasMachineDeployments;
```
Comment (maciaszczykm) [inline src/app/cluster/cluster-details/constraints/component.ts:105]: What will happen if there is no data, cluster is running but has no machine deployments? I think due to the new checks you have added not all scenarios are covered. `hasNoData` will return false and the zero state message will not be displayed.

Comment (kgroschoff) [reply src/app/cluster/cluster-details/constraints/component.ts]: Sorry, I've missed your comment.

Yes you are right. I thought about all cases again, and I think the machine deployments check isn't necessary at all for this views. data + cluster running is the important part and it doesn't change anything if there are machine deployments or not 👍

**Comment by maciaszczykm** [comment]
Looks good now, thanks.

/lgtm

**Comment by kgroschoff** [comment]
No problem, you're hint was right :)

**Pull request merged**